apps/server/src/modules/auth/next-auth.controller.ts

```diff
@@ -179,7 +179,7 @@ export class NextAuthController {
       });
       if (
         !req.headers?.referer ||
-        req.headers.referer.startsWith('https://accounts.google.com')
+        new URL(req.headers.referer).host === 'https://accounts.google.com'
```

Comment:
`new URL('https://accounts.google.com').host` === 'accounts.google.com'.
```suggestion
        new URL(req.headers.referer).origin === 'https://accounts.google.com'
```